src/MerchantCenter/MerchantCenterService.php

```diff
@@ -228,13 +229,19 @@ public function get_setup_status(): array {
 
 		$step = 'accounts';
 		if ( $this->connected_account() ) {
-			$step = 'product_listings';
+			$step = 'grant_rest_api_access';
 
-			if ( $this->saved_target_audience() && $this->saved_shipping_and_tax_options() ) {
-				$step = 'store_requirements';
+			$notifications_enabled = $this->container->get( NotificationsService::class )->is_enabled();
 
-				if ( $this->is_mc_contact_information_setup() && $this->checked_pre_launch_checklist() ) {
-					$step = 'paid_ads';
+			if ( ! $notifications_enabled || $this->is_rest_api_granted() ) {
```

Comment:
instead of `$this->is_rest_api_granted()` we might use `NotificationsService::is_ready()` that does the same so we can avoid adding more code doing the same.